go/pkg/services/logcollector/server.go

```diff
@@ -0,0 +1,605 @@
+	s.Logger.DebugCtx(ctx, "Initializing Server")
+
+	// start state updating goroutine
+	go s.stateStore.Initialize(ctx)
+
+	// start logging monitor
+	go s.monitorLogCollection(ctx)
+
+	return nil
+}
+
+func (s *LogCollectorServer) RegisterRoutes(ctx context.Context) {
+	s.AbstractMlrunGRPCServer.RegisterRoutes(ctx)
+	protologcollector.RegisterLogCollectorServer(s.Server, s)
+}
+
+// StartLog writes the log item info to the state file, gets the pod using the label selector,
+// triggers `monitorPod` and `streamLogs` goroutines.
+func (s *LogCollectorServer) StartLog(ctx context.Context, request *protologcollector.StartLogRequest) (*protologcollector.StartLogResponse, error) {
+
+	s.Logger.DebugWithCtx(ctx,
+		"Received Start Log request",
+		"RunUID", request.RunUID,
+		"Selector", request.Selector)
+
+	// check if the requested run uid's logs are already being collected
+	itemsInProgress, err := s.inMemoryState.GetItemsInProgress()
+	if err != nil {
+		err := errors.Wrap(err, "Failed to get items in progress from in memory state")
+		return &protologcollector.StartLogResponse{
+			Success: false,
+			Error:   err.Error(),
+		}, err
+	}
+	if _, running := itemsInProgress.Load(request.RunUID); running {
+		s.Logger.DebugWithCtx(ctx, "Logs are already being collected for this run uid", "runUID", request.RunUID)
+		return &protologcollector.StartLogResponse{
+			Success: true,
+		}, nil
+	}
+
+	var pods *v1.PodList
+
+	// list pods using label selector until a pod is found
+	errCount := 0
+	for pods == nil || len(pods.Items) == 0 {
+		pods, err = s.kubeClientSet.CoreV1().Pods(s.namespace).List(ctx, metav1.ListOptions{
+			LabelSelector: request.Selector,
+		})
+		if err != nil {
+
+			// we retry 3 times, as k8s might have some issues
+			if errCount <= 3 {
+				errCount++
+				s.Logger.WarnWithCtx(ctx, "Failed to list pods, retrying", "runUID", request.RunUID)
+			} else {
+
+				// fail on error
+				err := errors.Wrapf(err, "Failed to list pods for run id %s", request.RunUID)
+				return &protologcollector.StartLogResponse{
+					Success: false,
+					Error:   err.Error(),
+				}, err
+			}
+		}
+	}
+
+	// found a pod. for now, we only assume each run has a single pod.
+	pod := pods.Items[0]
+
+	// write log item in progress to state store
+	if err := s.stateStore.AddLogItem(ctx, request.RunUID, request.Selector); err != nil {
+		err := errors.Wrapf(err, "Failed to add run id %s to state file", request.RunUID)
+		return &protologcollector.StartLogResponse{
+			Success: false,
+			Error:   err.Error(),
+		}, err
+	}
+
+	// create a child context before calling goroutines, so it won't be canceled
+	logStreamCtx, cancelCtxFunc := mlruncontext.NewDetachedWithCancel(ctx)
+	startedStreamingGoroutine := make(chan bool, 1)
+
+	// stream logs to file
+	go s.startLogStreaming(logStreamCtx, request.RunUID, pod.Name, startedStreamingGoroutine, cancelCtxFunc)
+
+	// add Item to in-memory state
```

Comment:
```suggestion
	// add item to in-memory state
```

just fixed your upper -> lower. that being said, not sure I understand what "Item" is? can you explain further what/why you are doing